frontend/http/frontend.go

```diff
@@ -104,31 +113,13 @@ func (t *Frontend) handler() http.Handler {
 // ListenAndServe listens on the TCP network address t.Addr and blocks serving
 // BitTorrent requests until t.Stop() is called or an error is returned.
 func (t *Frontend) ListenAndServe() error {
-	t.s = &http.Server{
-		Addr:         t.Addr,
-		Handler:      t.handler(),
-		ReadTimeout:  t.ReadTimeout,
-		WriteTimeout: t.WriteTimeout,
-		ConnState: func(conn net.Conn, state http.ConnState) {
-			switch state {
-			case http.StateNew:
-				//stats.RecordEvent(stats.AcceptedConnection)
-
-			case http.StateClosed:
-				//stats.RecordEvent(stats.ClosedConnection)
-
-			case http.StateHijacked:
-				panic("http: connection impossibly hijacked")
-
-			// Ignore the following cases.
-			case http.StateActive, http.StateIdle:
-
-			default:
-				panic("http: connection transitioned to unknown state")
-			}
-		},
+	l, err := net.Listen("tcp", t.Addr)
+	if err != nil {
+		return err
 	}
-	t.s.SetKeepAlivesEnabled(false)
+
+	mux := cmux.New(l)
```

Comment:
Because of the weird behaviour with reads apparently timing out for you. If I'm not mistaken, otherwise the `SetReadTimeout` would have an effect, right?